packages/back-end/src/models/EventModel.ts

```diff
@@ -0,0 +1,116 @@
+import z from "zod";
+import _ from "lodash";
+import mongoose from "mongoose";
+import {
+  NotificationEventName,
+  notificationEventNames,
+  NotificationEventPayload,
+  NotificationEventResource,
+  notificationEventResources,
+} from "../events/base-types";
+import { EventInterface } from "../../types/event";
+import { errorStringFromZodResult } from "../util/validation";
+import { logger } from "../util/logger";
+
+const eventSchema = new mongoose.Schema({
+  id: {
+    type: String,
+    unique: true,
+    required: true,
+  },
+  dateCreated: Date,
+  organizationId: {
+    type: String,
+    required: true,
+  },
+  data: {
+    type: Object,
+    required: true,
+    validate: {
+      validator(value: unknown) {
+        // NotificationEventPayload<EventName, ResourceType, DataType>
+        const zodSchema = z
+          .object({
+            organization_id: z.string().optional(),
+            event_id: z.string().startsWith("event-"),
+            event: z.enum(notificationEventNames),
+            object: z.enum(notificationEventResources),
+            data: z.any(),
+          })
+          .strict();
+
+        const result = zodSchema.safeParse(value);
+
+        if (!result.success) {
+          const errorString = errorStringFromZodResult(result);
+          logger.error("Invalid Event data ", errorString);
+        }
+
+        return result.success;
+      },
+      message:
+        "Data is invalid. Must be of type NotificationEventPayload<NotificationEventName, NotificationEventResource, any>",
+    },
+  },
+});
+
+eventSchema.index({ organizationId: 1 });
```

Comment:
Wondering if we should do a compound index on `organizationId` (ascending) and `dateCreated` (descending).  I think most of the time, we'll want to fetch the most recent X events.

Reply:
I've made this change in the following commit: https://github.com/growthbook/growthbook/pull/745/commits/2acc72e9b1457ff1b9599c5c9e025b9e3a0921fe